src/lxc/criu.c

```diff
@@ -191,6 +191,8 @@ static void exec_criu(struct criu_opts *opts)
 		return;
 	}
 
+	static_args += 2 * cgroup_num_hierarchies();
```

Comment:
I don't see where you're checking for cgroup_num_hierarchie() == -1.  I do see that inside the while loop you'll try get_hierarchies() and get false, except you won't get there since the loop won't ever run since it goes from i = 0; i < -1.